WooCommerce/WooCommerceTests/ViewRelated/Products/SKU Scanner/BarcodeSKUScannerProductFinderTests.swift

```diff
@@ -66,16 +66,20 @@ final class BarcodeSKUScannerProductFinderTests: XCTestCase {
         let returningProduct = Product.fake()
         stores.whenReceivingAction(ofType: ProductAction.self, thenCall: { action in
             switch action {
-            case .retrieveFirstProductMatchFromSKU(_, _, let onCompletion):
-                onCompletion(.success(returningProduct))
+            case .retrieveFirstItemMatchFromSKU(_, _, let onCompletion):
+                onCompletion(.success(.product(returningProduct)))
             default:
                 break
             }
         })
 
         // When
         let scannedBarcode = ScannedBarcode(payloadStringValue: "123456", symbology: .aztec)
-        let retrievedProduct = try? await sut.findProduct(from: scannedBarcode, siteID: 1, source: source)
+        let result = try? await sut.searchBySKU(from: scannedBarcode, siteID: 1, source: source)
```

Comment:
Should we mark this method as `async throws` and the `result` as non-optional `try`? I'm wondering if this happens to fail we would just suppress the error and not know if we happen to change the following XCTFail check.

```suggestion
        let result = try await sut.searchBySKU(from: scannedBarcode, siteID: 1, source: source)
        XCTAssertNotNil(result, "Precondition: Result must not be nil")
```

Reply:
With our approach currently, if the result is nil it would fail on this assert:

```
 guard case let .product(retrievedProduct) = result else {
            return XCTFail("It didn't provide a product as expected")
        }
```

With the proposed approach we would add another assert, which in my opinion is not necessary because it's contained already there.